### DISCUSSION TITLE
What's the expected behaviour of extract_*****(true) in visual mode?

### DISCUSSION THREAD

**Discussion by olatheander** [mfussenegger/nvim-jdtls#292] [Q&A]
Hi,

I'm experimenting a bit with nvim-jdtls and got it working but not sure that I understand the expected behavior of the extract mapping e.g. 
```
vnoremap crv <Esc><Cmd>lua require('jdtls').extract_variable(true)<CR>`
```
Doing an extract operation in normal mode i.e. having the `crv` mapping to call `extract_variable()` (no param) I get a selection of feasible extractions from which to pick. Calling `extract_variable(true)` omits showing proposals, thus I assume that in visual mode I'm supposed to mark the block to extract but when I try nothing happens. Some hints how to operate in visual mode would be greatly appreciated.

Thanks

**Comment by mfussenegger** [comment]
> Calling extract_variable(true) omits showing proposals, thus I assume that in visual mode I'm supposed to mark the block

This is correct.
See:

https://user-images.githubusercontent.com/38700/180173697-5aac2529-8f5f-4f60-a3d5-156d96219d31.mp4
